backend/app/Actions/Teams/InviteTeamMemberAction.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Teams;
+
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\Hash;
+use App\Notifications\TeamMemberInvited;
+use App\Repositories\Contracts\WebsiteRepository;
+use App\Repositories\Contracts\UserRepository;
+use App\Entities\User;
+
+final class InviteTeamMemberAction
+{
+    private $userRepository;
+    private $websiteRepository;
+
+    public function __construct(
+        UserRepository $userRepository,
+        WebsiteRepository $websiteRepository
+    ) {
+        $this->userRepository = $userRepository;
+        $this->websiteRepository = $websiteRepository;
+    }
+
+    public function execute(InviteTeamMemberRequest $request): void
+    {
+        $website = $this->websiteRepository->getById($request->websiteId());
+        $password = '';
+        $teamMember = $this->userRepository->getByEmail($request->email());
+
+        if (!$teamMember) {
+            $password = Str::random(8);
```

Comment:
I suppose, there should be a whole registration process because the user also should confirm an account. Use RegisterAction here

Reply:
Fixed